src/datasources/db/v2/database-initialize.hook.ts

```diff
@@ -0,0 +1,16 @@
+import { PostgresDatabaseService } from '@/datasources/db/v2/postgres-database.service';
+import { Inject, Injectable, type OnModuleInit } from '@nestjs/common';
+
+@Injectable()
+export class DatabaseInitializeHook implements OnModuleInit {
+  public constructor(
+    @Inject(PostgresDatabaseService)
+    private readonly postgresDatabaseService: PostgresDatabaseService,
+  ) {
+    //
```

Comment:
```suggestion
```

Reply:
I've commented this out because the getRepository method in postgres-database.service initializes the connection if it hasn't been initialized yet. This ensures the connection is only initialized once when a call to the database is made.

Do you think we should uncomment it or remove the hook? Uncommenting won’t reinitialize the connection, but it does create a redundant call.

Reply:
Done
https://github.com/safe-global/safe-client-gateway/commit/2ea0e5d8a752fd31cff745ff3490d8fe8134ddc7

Reply:
Sorry for the confusion, somehow the change was reverted on my end. Done
https://github.com/safe-global/safe-client-gateway/pull/1988/commits/e907e3cc5db5e7b605fce8afb51ea6464a464252